src/java.base/share/native/libjava/ClassLoader.c

```diff
@@ -228,21 +233,25 @@ Java_java_lang_ClassLoader_defineClass0(JNIEnv *env,
 
     if (data == NULL) {
         JNU_ThrowNullPointerException(env, 0);
-        return 0;
+        return NULL;
     }
 
     /* Work around 4153825. malloc crashes on Solaris when passed a
      * negative size.
      */
     if (length < 0) {
         JNU_ThrowArrayIndexOutOfBoundsException(env, 0);
-        return 0;
+        return NULL;
     }
 
-    body = (jbyte *)malloc(length);
-    if (body == 0) {
+    // On malloc(0), implementators of malloc(3) have the choice to return either
+    // NULL or a unique non-NULL pointer. To unify libc behavior across our platforms
+    // we chose the latter. (see 8283225)
+    body = (jbyte *)malloc(length < 1 ? 1 : length);
```

Comment:
The comment talks about a length of zero, but the code does something else with length < 1.
I'd rather not see negative sizes conflated with zero.